### PR TITLE
refactor(efb): improvements to code quality and conformance to eslint-config v0.2.0 rules

### PR DESCRIPTION
**Pull request by ErickSharp**
## Summary of Changes

remove class components, replace with function components and conform to the new eslint config v.0.2.0's rules; meaning that I removed the entirety of errors occurring within the EFB's code, in addition to a large amount of warnings. 

## Additional context

Discord username (if different from GitHub): Erick [Z-6]#6484

## Testing instructions

This pull request just refactors some code in the EFB, just ensure that nothing breaks. 

## How to download the PR for QA

Every new commit to this PR will cause a new A32NX artifact to be created, built, and uploaded.

1. Make sure you are signed in to GitHub
1. Click on the **Checks** tab on the PR
1. On the left side, click on the bottom **PR** tab
1. Click on the **A32NX** download link at the bottom of the page

### CODE REVIEW

File: src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx
Comment (ZigTag) [inline src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx:66]: Weird wrapping, move the brace to new line and move the other code after it

**Review by ZigTag — APPROVED** [review]
LGTM

**Review by ZigTag — DISMISSED** [review]
LGTM, just needs to fix this inconsistent formatting

File: src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx
Comment (ZigTag) [inline src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx:66]: Remove Space

**Comment by Lucky38i** [comment]
I'm not sure this is out of the scope of this PR but it might be useful to resolve the misuse of react hooks as-well as pointed out in #5365

**Review by Lucky38i — COMMENTED** [review]
LGTM. However I'm a little confused. In Help.tsx you state the React.FC type as-well as destructure the props. However for all other files, you infer the the React Type for all the components and just have a `props` parameter in the component parameter definition.

Personally I like the way it's done in Help.tsx. May I suggest we fully make use of TS and define all the types for each component `Foo: React.FC<Props> = ({...destructredProps})` for improved consistency and type safety. 

I'm leaving this as a comment for now instead of approval or request changes, to see what others think.

**Comment by Benjozork** [comment]
It would definitely help if this could address some of the ESLint errors in #5365

**Comment by ErickSharp** [comment]
stop I think I just broke the branch 💀

**Comment by ErickSharp** [comment]
@Lucky38i hey Lucky, do you think I can get another review, I just rebased to fix some merge conflicts and need your eyes again.

File: src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Components/Form/SelectInput/SelectInput.tsx:57]: Hm. @ErickSharp as we discussed on Discord you can simply name the function `handleClick`. I would not inline it, harder to read IMO.

**Comment by Benjozork** [comment]
I will test that this is compatible with the new eslint config before merging

**Comment by ErickSharp** [comment]
> I will test that this is compatible with the new eslint config before merging

ok, I fixed the majority of errors

File: src/instruments/src/EFB/ChartsApi/ChartFox.tsx
Comment (Lucky38i) [inline src/instruments/src/EFB/ChartsApi/ChartFox.tsx:27]: Are we sure we want to be throwing an error rather than `console.error()` ? Throwing an error stops execution. However someone more experienced in error handling can say what's the better choice.

File: src/instruments/src/EFB/Efb.tsx
Comment (Lucky38i) [inline src/instruments/src/EFB/Efb.tsx:165]: why were these logs removed? they seem useful for debugging and wont' clog the console.

File: src/instruments/src/EFB/Dispatch/Dispatch.tsx
Comment (Lucky38i) [inline src/instruments/src/EFB/Dispatch/Dispatch.tsx:59]: since we're not using `this` anymore maybe switch these to arrow functions? Functionality remains the same though so it's not necessary.

File: src/instruments/src/EFB/Performance/Widgets/LandingWidget.tsx
Comment (Lucky38i) [inline src/instruments/src/EFB/Performance/Widgets/LandingWidget.tsx:111]: Why does this return a promise?

**Review by Lucky38i — CHANGES_REQUESTED** [review]
Just some suggested changes and open discussion

File: src/instruments/src/EFB/Dispatch/Dispatch.tsx
Comment (ErickSharp) [reply src/instruments/src/EFB/Dispatch/Dispatch.tsx:59]: these can remain as-is

File: src/instruments/src/EFB/ChartsApi/ChartFox.tsx
Comment (ErickSharp) [reply src/instruments/src/EFB/ChartsApi/ChartFox.tsx:27]: Preferably, these would not be console.logs and actual thrown Errors that could be handled when called. I haven't gotten back to this PR but I will need to implement the catching of the exception.

File: src/instruments/src/EFB/Efb.tsx
Comment (ErickSharp) [reply src/instruments/src/EFB/Efb.tsx:165]: I don't think it is necessary to annotate certain operations being done, and anything that goes wrong whilst these operations happen would preferably throw an error that could be handled and that would be the primary way of debugging in the event that something went wrong.

File: src/instruments/src/EFB/Performance/Widgets/LandingWidget.tsx
Comment (ErickSharp) [reply src/instruments/src/EFB/Performance/Widgets/LandingWidget.tsx:111]: Because it is performing an asynchronous operation and it does not resolve to a value since there is nothing to be returned from what the body of the function is intended to do.

Comment (tracernz) [reply src/instruments/src/EFB/Performance/Widgets/LandingWidget.tsx:111]: > Why does this return a promise? It doesn't look to be handled.

An async function _always_ returns a promise.

**Comment by Benjozork** [comment]
As we discussed on call yesterday did you fix the `useEffect` dependency errors present in eslint020 ?

Also, the current lint is failing.

**Comment by Benjozork** [comment]
If you don't have time to fix the eslint020 errors, we can merge this without that and do it in another PR. I don't want to hold this up.

**Comment by ErickSharp** [comment]
> As we discussed on call yesterday did you fix the `useEffect` dependency errors present in eslint020 ?
> 
> Also, the current lint is failing.

All that remains are warnings, there are currently no eslint-reported errors within the EFB's codebase.

**Review by Lucky38i — APPROVED** [review]

**Comment by ErickSharp** [comment]
This pull request can be re-marked for v0.7.0, all that is left for it to do is go through QA testing.

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5541
Tier of Testing: 1
Date: 21/08/2021

Testing Process:
- Loaded in on runway at EGSS: 40fps.
- Turned on EFB: decreased to 17FPS.
- Reloaded sim, loaded same spot: 40FPS.
- Turned on EFB: decreased to 17FPS again.

Removed PR and installed dev version:
- Loaded in on runway at EGSS: 40fps.
- Turned on EFB: remained at 39-40fps.

Negatives: Not sure if just me, massive performance impact on PR. Did not continue with testing.
Testing Results: FAIL

Conclusions: As above, not sure if a localised issue but PR introduced significant performance loss.

**Comment by FlightSimmer7700** [comment]
> 
> 
> Quality Assurance Tester Report
> 
> Discord: █▀█ █▄█ ▀█▀#2123
> Object of testing: #PR 5541
> Tier of Testing: 1
> Date: 21/08/2021
> 
> Testing Process:
> 
>     * Loaded in on runway at EGSS: 40fps.
> 
>     * Turned on EFB: decreased to 17FPS.
> 
>     * Reloaded sim, loaded same spot: 40FPS.
> 
>     * Turned on EFB: decreased to 17FPS again.
> 
> 
> Removed PR and installed dev version:
> 
>     * Loaded in on runway at EGSS: 40fps.
> 
>     * Turned on EFB: remained at 39-40fps.
> 
> 
> Negatives: Not sure if just me, massive performance impact on PR. Did not continue with testing.
> Testing Results: FAIL
> 
> Conclusions: As above, not sure if a localised issue but PR introduced significant performance loss.

Same issue reproduced on my end
EFB off:
image
EFB on: 
image

**Comment by ErickSharp** [comment]
> Quality Assurance Tester Report
> 
> Discord: █▀█ █▄█ ▀█▀#2123
> Object of testing: #PR 5541
> Tier of Testing: 1
> Date: 21/08/2021
> 
> Testing Process:
> 
> * Loaded in on runway at EGSS: 40fps.
> * Turned on EFB: decreased to 17FPS.
> * Reloaded sim, loaded same spot: 40FPS.
> * Turned on EFB: decreased to 17FPS again.
> 
> Removed PR and installed dev version:
> 
> * Loaded in on runway at EGSS: 40fps.
> * Turned on EFB: remained at 39-40fps.
> 
> Negatives: Not sure if just me, massive performance impact on PR. Did not continue with testing.
> Testing Results: FAIL
> 
> Conclusions: As above, not sure if a localised issue but PR introduced significant performance loss.

Likely too many re-renders, can you speak on what pages show an even greater performance loss than others?

**Comment by ErickSharp** [comment]
On further review, I don't see anything that can be leading to this severe of a performance drop. I will test locally and try and make some changes and see if anything changes following them.

**Comment by ErickSharp** [comment]
@NUTTTTTTTT @FlightSimmer7700 Did you both use the artifact or checkout and build the PR?

**Comment by ErickSharp** [comment]
@NUTTTTTTTT @FlightSimmer7700 I tried testing some things out and revealed the issue was being caused by too many renders from the StatusBar component, which is why every page was slow. The issue is now fixed, please re-test and let me know of any issues that have come up if any at all!

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5541
Tier of Testing: 1
Date: 21/08/2021

Testing Process:
- Loaded in on runway at EGSS: 40fps.
- Turned on EFB: remained at or around 40fps, performance impact resolved.
- Went through functions, all seemed to be working except as soon as clicking into calibration menu, EFB goes black and needs a sim/flight restart to bring it back.

Negatives: Breaks EFB when trying to calibrate.
Results: FAIL

Media: https://youtu.be/f78Osq3iwJQ

**Comment by NUTTTTTTTT** [comment]
Quality Assurance Tester Report

Discord: █▀█ █▄█ ▀█▀#2123
Object of testing: #PR 5541
Tier of Testing: 1
Date: 21/08/2021

Testing Process:

Repeat of previous testing process for consistency.
- Loaded in on runway at EGSS: 40fps.
- Turned on EFB: remained at or around 40fps, performance impact resolved still.
- Went through functions, all seemed to be working okay now, no issues observed with calibration and all menus seem operational in testing.

Negatives: N/A
Results: PASS

Conclusions: No longer any issues with EFB, all works exactly as before which is intended, so good stuff!

File: src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleSimVar.tsx
Comment (Benjozork) [inline src/instruments/src/EFB/Settings/ThrottleConfig/ThrottleSimVar.tsx]: You cannot use react hooks outside of react components

**Comment by oim1** [comment]
Quality Assurance Trainee Report

Object of testing: #5541
Tier of Testing: 1
Date: (22/8/2021)

Testing Process: 
Loaded cold and dark at YMML. 
Turned on flypad. 
Tested each feature to ensure they were working properly. 

Negatives: N/A
Testing Results: Passed

Conclusions: I managed to get 58fps when looking at the flypad so performance isn't an issue. Everything works fine. LGTM.

**Comment by NUTTTTTTTT** [comment]
Was hoping to get some more eyes on this from QA before merging given the previous issues in other tests, but seems to have merged while I was sleeping, this still has issues. https://github.com/flybywiresim/a32nx/issues/5698 now introduced into dev. Tested and verified on my end as well - same issue persists there - issue with both V/S and FPA.

**Pull request merged**
